explorer/models.py

```diff
@@ -160,15 +164,51 @@ def __init__(self, sql, title=None,is_connection_type_pii=None):
         cursor, duration = self.execute_query()
 
         self._description = cursor.description or []
-        self._data = [list(r) for r in cursor.fetchall()]
 
-        self.duration = duration
+        if not is_connection_type_pii:
+            type_code_and_column_indices_to_be_masked_dict = self.get_type_code_and_column_indices_to_be_masked_dict()
+            self._data = self.get_data_to_be_displayed(cursor, type_code_and_column_indices_to_be_masked_dict)
+        else:
+            self._data = [list(r) for r in cursor.fetchall()]
 
+        self.duration = duration
         cursor.close()
 
         self._headers = self._get_headers()
         self._summary = {}
 
+    def get_type_code_and_column_indices_to_be_masked_dict(self):
+        type_code_and_column_indices_to_be_masked_dict = {
+            TYPE_CODE_FOR_JSON: [],
+            TYPE_CODE_FOR_TEXT: []
+        }
+        # Collect the indices for JSON and text columns
+        for index, column in enumerate(self._description):
+            if hasattr(column, "type_code") and column.type_code in type_code_and_column_indices_to_be_masked_dict:
+                type_code_and_column_indices_to_be_masked_dict[column.type_code].append(index)
+
+        return type_code_and_column_indices_to_be_masked_dict
+
+    @staticmethod
+    def get_data_to_be_displayed(cursor, type_code_and_column_indices_to_be_masked_dict):
+        data_to_be_displayed = []
+        for row in cursor.fetchall():
+            modified_row = list(row)
+            for type_code, indices in type_code_and_column_indices_to_be_masked_dict.items():
+                for index in indices:
+                    if not modified_row[index]:
+                        continue
+                    if type_code == TYPE_CODE_FOR_JSON:
```

Comment:
Too much nesting, also please write the docstring k what this method is doing and what would be there in each fields with example so that anyone can understand it easily

Reply:
Resovled.